static/js/hotkey.js

```diff
@@ -874,13 +875,9 @@ export function process_hotkey(e, hotkey) {
             return true;
     }
 
-    // We don't want hotkeys below this to work when recent topics is
-    // open. These involve hotkeys that can only be performed on a message.
-    if (recent_topics_util.is_visible()) {
-        return false;
-    }
-
-    if (message_lists.current.empty()) {
+    // We don't want hotkeys below this to work when message feed is
+    // not visible. These involve hotkeys that can only be performed on a message.
+    if (message_lists.current.empty() || !narrow_state.is_message_feed_visible()) {
```

Comment:
I'm reversing the order of these checks; logically, we want to check if the message feed is visible before inspecting `message_lists.current`, since the current message list is effectively undefined if no message feed is visible.

Probably there's a refactor we could do to make sure everything that interacts with `message_lists.current` is guarded by a `is_message_feed_visible` check.